### ISSUE TITLE
Experimental annotation rule is broken with file annotations 

### ISSUE THREAD

**Issue by jdemeulenaere** [pinterest/ktlint#747]
## Expected Behavior

Running `ktlint --experimental` on the following file should succeed:

```kotlin
@file:SomeFileAnnotation()

class Test
```

## Observed Behavior

Fails with `Missing spacing after @file:SomeFileAnnotation()`

## Steps to Reproduce

```bash
$ cat << EOF > /tmp/Test.kt
> @file:SomeFileAnnotation()
>
> class Test
> EOF

$ ktlint --experimental /tmp/Test.kt
/private/tmp/Test.kt:1:26: Missing spacing after @file:SomeFileAnnotation()
```

## Your Environment
* ktlint 0.36.0
* macOS 10.15.4

**Comment by romtsn** [comment]
Will be closed by #714 I believe
